packages/paste-core/components/progress-steps/src/types.ts

```diff
@@ -0,0 +1,26 @@
+import type {BoxProps} from '@twilio-paste/box';
+
+interface ProgressStepProps extends Omit<React.HTMLAttributes<HTMLElement>, 'color'> {
+  element?: BoxProps['element'];
+  as?: 'button' | 'a' | 'div';
+  href?: string;
+  onClick?: () => void;
+}
```

Comment:
suggestion: an improvement here would be to follow the Badge typings where we conditionally type the html element based on the as prop. So if it's a button it extends button, if it's a link it extends anchor props. https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/badge/src/types.ts#L26